docs/features/incremental-generators.md

```diff
@@ -14,6 +14,40 @@ strategies that can be applied in a high performance way by the hosting layer.
 - Support generating more items that just source texts
 - Exist alongside `ISourceGenerator` based implementations
 
+## Simple Example
+
+We begin by defining a simple incremental generator that extracts the contents
+of additional text files and makes their contents available as compile time
+`const`s. In the following section we'll go into more depth around the concepts
+shown.
+
+```csharp
+[Generator]
+public class Generator : IIncrementalGenerator
+{
+    public void Initialize(IncrementalGeneratorInitializationContext initContext)
+    {
+        // define the execution pipeline here via a series of transformations:
+
+        // find all additional files that end with .txt
+        IncrementalValuesProvider<AdditionalText> textFiles = context.AdditionalTextsProvider.Where(static file => file.Path.EndsWith(".txt"));
```

Comment:
nit: consider either removing 'static' modifier on lambda or including it for all lambdas where it would be legal